pkg/bytecode/vm.go

```diff
@@ -75,13 +82,12 @@ func (vm *VM) lastPoppedStackElem() value {
 	return vm.stack[vm.sp]
 }
 
-func (vm *VM) push(o value) error {
+func (vm *VM) push(o value) {
 	if vm.sp >= StackSize {
-		return ErrStackOverflow
+		panic(ErrStackOverflow)
```

Comment:
This is a "user" error (should wrap `ErrPanic` or similar) and should therefore be returned imo.

Reply:
@Rossiar , I think this is unresolved -`ErrStackOverflow` is a "user error" and should be returned rather than panic'ed with - agreed?

Reply:
Sorry I forgot to comment here - how about a recover in `Run`? This keeps the code less cluttered and also collects index out of bounds and other unforeseen panics, and avoids the user seeing a wall of stack trace as an error

Reply:
Yes, personally I really like this approach within a package, however unfortunately tinygo-wasm does not support recover. It's super annoying.
So recover is not an option at this stage I'm afraid :(

https://github.com/tinygo-org/tinygo/issues/2914

Reply:
Ah what a shame, in that case I will put the error handling back

Reply:
tbh - panics are not handled at all on evy web: https://github.com/evylang/todo/issues/68 I really need to sort this out.